operator/v1/types_network.go

```diff
@@ -0,0 +1,189 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Network describes the cluster's desired network configuration. It is
+// consumed by the cluster-network-operator.
+// +k8s:openapi-gen=true
+type Network struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   NetworkSpec   `json:"spec,omitempty"`
+	Status NetworkStatus `json:"status,omitempty"`
+}
+
+// NetworkStatus is currently unused. Instead, status
+// is reported in the Network.config.openshift.io object.
+type NetworkStatus struct {
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NetworkList contains a list of Network configurations
+type NetworkList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Network `json:"items"`
+}
+
+// NetworkSpec is the top-level network configuration object.
+type NetworkSpec struct {
+	// clusterNetwork is the IP address pool to use for pod IPs.
+	// Some network providers, e.g. OpenShift SDN, support multiple ClusterNetworks.
+	// Others only support one. This is equivalent to the cluster-cidr.
+	ClusterNetwork []ClusterNetworkEntry `json:"clusterNetwork"`
+
+	// serviceNetwork is the ip address pool to use for Service IPs
+	// Currently, all existing network providers only support a single value
+	// here, but this is an array to allow for growth.
+	ServiceNetwork []string `json:"serviceNetwork"`
+
+	// defaultNetwork is the "default" network that all pods will receive
+	DefaultNetwork DefaultNetworkDefinition `json:"defaultNetwork"`
+
+	// additionalNetworks is a list of extra networks to make available to pods
+	// when multiple networks are enabled.
+	AdditionalNetworks []AdditionalNetworkDefinition `json:"additionalNetworks,omitempty"`
+
+	// disableMultiNetwork specifies whether or not multiple pod network
+	// support should be disabled. If unset, this property defaults to
+	// 'false' and multiple network support is enabled.
+	DisableMultiNetwork *bool `json:"disableMultiNetwork,omitempty"`
+
+	// deployKubeProxy specifies whether or not a standalone kube-proxy should
+	// be deployed by the operator. Some network providers include kube-proxy
+	// or similar functionality. If unset, the plugin will attempt to select
+	// the correct value, which is false when OpenShift SDN and ovn-kubernetes are
+	// used and true otherwise.
+	// +optional
+	DeployKubeProxy *bool `json:"deployKubeProxy,omitempty"`
```

Comment:
I've talked with @squeed about this being pointer and he presented reasonable arguments for that. So I'll leave it as a pointer.